Userland/Libraries/LibWeb/HTML/Navigation.cpp

```diff
@@ -322,6 +323,63 @@ WebIDL::ExceptionOr<NavigationResult> Navigation::reload(NavigationReloadOptions
     return navigation_api_method_tracker_derived_result(api_method_tracker);
 }
 
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigation-traverseto
+WebIDL::ExceptionOr<NavigationResult> Navigation::traverse_to(String key, NavigationOptions const& options)
+{
+    auto& realm = this->realm();
+    // The traverseTo(key, options) method steps are:
+
+    // 1. If this's current entry index is −1, then return an early error result for an "InvalidStateError" DOMException.
+    if (m_current_entry_index == -1)
+        return early_error_result(WebIDL::InvalidStateError::create(realm, "Cannot traverseTo: no current session history entry"));
+
+    // 2. If this's entry list does not contain a NavigationHistoryEntry whose session history entry's navigation API key equals key,
+    //    then return an early error result for an "InvalidStateError" DOMException.
+    auto it = m_entry_list.find_if([&key](auto const& entry) {
+        return entry->session_history_entry().navigation_api_key == key;
+    });
+    if (it == m_entry_list.end())
+        return early_error_result(WebIDL::InvalidStateError::create(realm, "Cannot traverseTo: key not found in session history list"));
+
+    // 3. Return the result of performing a navigation API traversal given this, key, and options.
+    return perform_a_navigation_api_traversal(key, options);
+}
+
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#performing-a-navigation-api-traversal
+WebIDL::ExceptionOr<NavigationResult> Navigation::back(NavigationOptions const& options)
+{
+    auto& realm = this->realm();
+    // The back(options) method steps are:
+
+    // 1. If this's current entry index is −1 or 0, then return an early error result for an "InvalidStateError" DOMException.
+    if (m_current_entry_index == -1 || m_current_entry_index == 0)
+        return early_error_result(WebIDL::InvalidStateError::create(realm, "Cannot navigate back: no previous session history entry"));
+
+    // 2. Let key be this's entry list[this's current entry index − 1]'s session history entry's navigation API key.
+    auto key = m_entry_list[m_current_entry_index - 1]->session_history_entry().navigation_api_key;
+
+    // 3. Return the result of performing a navigation API traversal given this, key, and options.
+    return perform_a_navigation_api_traversal(key, options);
+}
+
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigation-forward
+WebIDL::ExceptionOr<NavigationResult> Navigation::forward(NavigationOptions const& options)
+{
+    auto& realm = this->realm();
+    // The forward(options) method steps are:
+
+    // 1. If this's current entry index is −1 or is equal to this's entry list's size − 1,
+    //    then return an early error result for an "InvalidStateError" DOMException.
+    if (m_current_entry_index == -1 || m_current_entry_index == static_cast<i64>(m_entry_list.size()))
```

Comment:
"equal to this's entry list's size − 1" looks like you missed -1 part

Reply:
do'h I sure did! fixed now